src/OrleansSQLUtils/Storage/IRelationalStorage.cs

```diff
@@ -57,13 +60,15 @@ public interface IRelationalStorage
         ///}).ConfigureAwait(continueOnCapturedContext: false);                
         /// </code>        
         /// </example>
-        Task<IEnumerable<TResult>> ReadAsync<TResult>(string query, Action<IDbCommand> parameterProvider, Func<IDataRecord, int, TResult> selector);
+        Task<IEnumerable<TResult>> ReadAsync<TResult>(string query, Action<IDbCommand> parameterProvider, Func<IDataRecord, int, CancellationToken, TResult> selector, CancellationToken cancellationToken = default(CancellationToken), CommandBehavior commandBehabior = CommandBehavior.Default);
```

Comment:
I added these as default arguments, because:
1) These are the implicit defaults in ADO.NET and likely never change.
2) To avoid multiple overloaded functions, in conjunction with the observation on 1).

I think `CancellationToken` will be used more frequently, so that is why it is before `CommandBehavior`.